bot/src/keeper/surplus.ts

```diff
@@ -0,0 +1,131 @@
+import BigNumber from 'bignumber.js';
+import { SurplusAuctionActive } from 'auctions-core/src/types';
+import { bidToSurplusAuction, enrichSurplusAuction, collectSurplusAuction } from 'auctions-core/src/surplus';
+import getSigner from 'auctions-core/src/signer';
+import { fetchAllowanceAmountMKR, setAllowanceAmountMKR } from 'auctions-core/src/authorizations';
+import { fetchBalanceMKR } from 'auctions-core/src/wallet';
+import { formatToAutomaticDecimalPointsString } from 'auctions-core/src/helpers/formatToAutomaticDecimalPoints';
+import { KEEPER_SURPLUS, KEEPER_SURPLUS_MINIMUM_NET_PROFIT_DAI } from '../variables';
+import { isSetupCompleted } from './setup';
+
+const currentlyExecutedAuctions = new Set();
+
+const checkAndParticipateIfPossible = async function (network: string, auction: SurplusAuctionActive) {
+    // check if setupKeeper hasn't run
+    if (!isSetupCompleted) {
+        return;
+    }
+
+    if (!KEEPER_SURPLUS || Number.isNaN(KEEPER_SURPLUS_MINIMUM_NET_PROFIT_DAI)) {
+        return;
+    }
+
+    const signer = await getSigner(network);
+    const walletAddress = await signer.getAddress();
+
+    // enrich the auction with more numbers
+    const auctionTransaction = await enrichSurplusAuction(network, auction);
+
+    // check if auction became inactive or finished
+    if (auctionTransaction.state === 'ready-for-collection') {
+        console.info(`keeper: surplus auction "${auction.id}" has already finished`);
+        // check if the current user is the winner
+        if (auctionTransaction.receiverAddress === walletAddress) {
```

Comment:
This should be checked against the same case since it can (and it is in practice) have different cases stored on the chain and derived by `ethers` from the private key